Connected System Plug-in (CSP) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/googledrive/templates/GoogleDriveUploadFileIntegrationTemplate.java

```diff
@@ -0,0 +1,123 @@
+
+@TemplateId(name = "GoogleDriveUploadFileIntegrationTemplate")
+public class GoogleDriveUploadFileIntegrationTemplate extends SimpleIntegrationTemplate {
+
+  public static final String FILE_ID_KEY = "fileId";
+  public static final String FOLDER_ID_KEY = "folderID";
+
+  @Override
+  protected SimpleConfiguration getConfiguration(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      PropertyPath updatedProperty,
+      ExecutionContext executionContext) {
+    return integrationConfiguration.setProperties(
+        textProperty(FILE_ID_KEY)
+            .label("File ID")
+            .instructionText("This can be found in a shareable link for a file as part of the URL, e.g. " +
+                "https://drive.google.com/open?id=<FileID>")
+            .build(),
+        FolderPropertyDescriptor.builder()
+            .key(FOLDER_ID_KEY)
+            .label("Save to Folder")
+            .build()
+    );
+  }
+
+  @Override
+  protected IntegrationResponse execute(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      ExecutionContext executionContext) {
+    String fileId = integrationConfiguration.getValue(FILE_ID_KEY);
+    Long folderId = integrationConfiguration.getValue(FOLDER_ID_KEY);
+
+    IntegrationDesignerDiagnostic.IntegrationDesignerDiagnosticBuilder diagnosticBuilder = IntegrationDesignerDiagnostic
+        .builder();
+    Stopwatch stopwatch = Stopwatch.createStarted();
+
+    //GoogleCredential is used to create a Google client with Drive. To send a file, you need to supply a File
+    //and an InputStreamContent.
+    GoogleCredential credential = new GoogleCredential().setAccessToken(
+        executionContext.getAccessToken().get());
+    Drive drive = new Drive.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance(),
+        credential).build();
+
+    File fileMetadata;
+    OutputStream outputStream = new ByteArrayOutputStream();
+    try {
+      fileMetadata = drive.files().get(fileId).execute();
+      drive.files().get(fileId).executeMediaAndDownloadTo(outputStream);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    long elapsed = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+    Map<String,Object> requestDiagnostics = getRequestDiagnostics(connectedSystemConfiguration,
+        integrationConfiguration);
+
+    byte[] buffer = new byte[4096];
+    InputStream inputStream = new ByteArrayInputStream(buffer);
+    try {
+      IOUtils.copyLarge(inputStream, outputStream);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    String fileName = fileMetadata.getName();
+    Document document = executionContext.getDocumentDownloadService()
+        .downloadDocument(inputStream, folderId, fileName);
+    Map<String,Object> result = new HashMap<>();
+    result.put("File uploaded", document);
```

Comment:
is this the recommended behavior for the results map when you download a document? or do we not really have a recommended behavior?

Reply:
yes, I believe in the documentService class we have documented that we need them to put the document in the result map otherwise we will delete the uploaded doc